### PR TITLE
fix `Append the ecdhCurve = 'auto' option to request options if node version higher than 8.5` (close #1401)

### LINKED ISSUE #1401: Append the ecdhCurve = 'auto' option to request options if node version higher than 8.5
Based on [question](https://testcafe-discuss.devexpress.com/t/problems-with-x-frame-options/628).

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

File: src/utils/node-version.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export default (() => {
```
Comment (miherlosev) [inline src/utils/node-version.js]: Let's use the module from npm with same functionality: [node-verson](https://www.npmjs.com/package/node-version) for example.

File: src/request-pipeline/destination-request/index.js [hunk 2]
```diff
@@ -17,6 +18,10 @@ const SOCKET_HANG_UP_ERR_RE      = /socket hang up/i;
 const IS_DNS_ERR_MSG_RE          = /ECONNREFUSED|ENOTFOUND|EPROTO/;
 const IS_DNS_ERR_CODE_RE         = /ECONNRESET/;
 
+// NOTE: Node.js changes the default behavior of SSL starting from v8.6,
+// and we need to add an additional option to request options if this condition is true
```
Comment (miherlosev) [inline src/request-pipeline/destination-request/index.js]: Add link to an approprite issue

Comment (miherlosev) [reply src/request-pipeline/destination-request/index.js]: Change comment to.
``Starting from 8.6 version, Node.js changes behavior related with sending requests to sites using SSL2 and SSL3 protocol versions.
It affects the `https` core module and can break a proxying of some sites.
This is why, we are forced to use the special hack.
For details, see https://github.com/nodejs/node/issues/16196

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Pull request merged**